apps/o2-blocks/src/p2-autocomplete/editor.js

```diff
@@ -19,9 +19,9 @@ const COMMON_PREFIXES = /(team|a8c|woo|happiness)/i;
 const stripCommonWords = str => str.replace( COMMON_PREFIXES, ' ' );
 
 const p2s = apiFetch( {
-	path: '/internal/P2s',
-} ).then( result =>
-	map( result.list, ( p2, subdomain ) => {
+	path: '/internal/P2s?current_blog=' + window._currentSiteId + '&get_most_used=true',
+} ).then( result => {
+	result.list = mapValues( result.list, ( p2, subdomain ) => {
```

Comment:
Let's introduce a new `const` for this, I don't like to mutating a function arg too much, especially if it's an object property and there's not really any reason we'd need it this way :grimacing:

Reply:
How about sticking with lodash `map` (which returns an array of the mapped values) rather than `mapValues` (which returns an object, retaining keys)? It looks like we're only using `Object.values( result.list )` below and aren't really interested in the keys. If it's about the `pick`, we might be able to just pick by the `subdomain` property?

Reply:
> …a new const for this…

on it

> How about sticking with lodash map…

If we `map`, we will lose the ability to pick by subdomain directly. We can still get those items by `subdomain` prop as you suggest but that means doing a loop over all the items just to get ten of them which seems like an unnecessary overhead.

Reply:
I've converted everything into a single `reduce` call and removed the debounce which makes it much more responsive (it is no longer needed, since we avoid API search). What do you think?